File: Python/faculty.py
from person import Person

# Implemented Inheritance

class Faculty(Person):

    def __init__(self, id, name, email):
        super().__init__(id, name)
        self.__email = email

    def showRole(self):
        print(self.getName() + " is a Faculty")

    def teachCourse(self, c):
        print(self.getName() + " teaches " + c.name)

File: C#/Faculty.cs
using System;

namespace StudentManagementSystem
{
    // Implemented Inheritance
    public class Faculty : Person
    {
        private string email;

        public Faculty(int id, string name, string email)
            : base(id, name)
        {
            this.email = email;
        }

        public override void ShowRole()
        {
            Console.WriteLine(Name + " is a Faculty");
        }

        public void TeachCourse(Course c)
        {
            Console.WriteLine(Name + " teaches " + c.name);
        }
    }
}
